Embrace Unity iOS Interface/SpanRepository.swift

```diff
@@ -0,0 +1,92 @@
+//  Essentially identical to RN SpanRepository.Swift
+
+import Foundation
+import OSLog
+import OpenTelemetryApi
+
+// Should not get hit under normal circumstances, add as a guard against misinstrumentation
+private let MAX_STORED_SPANS = 10000
+
+class SpanRepository {
+    private let activeSpansQueue = DispatchQueue(label: "io.embrace.Unity.activeSpans", 
+                                                 attributes: .concurrent)
+    private let completedSpansQueue = DispatchQueue(label: "io.embrace.Unity.completedSpans", 
+                                                    attributes: .concurrent)
+    private var activeSpans = [String: Span]()
+    private var completedSpans = [String: Span]()
+    private var log = OSLog(subsystem: "Embrace", category: "SpanRepository")
```

Comment:
Just in case, `os_log` also logs things in Release Mode. I don't know if it's intended but just mentioning to prevent any kind of problems.
Is a bit more performant than `print` or `NSLog` tho; just to take it into consideration whenever we're dealing with private/public data

Reply:
That makes sense. Thank you for the note. I should probably look into some way to handle logging enablement in the future.